design/providers/kubernetes_secrets_storage_csi/conjur_csi_provider_solution_design.md

```diff
@@ -0,0 +1,314 @@
+
+A pod that wants to use these secrets would reference the `SecretProviderClass` in its volumes definition like so:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: mypod
+spec:
+  containers:
+  - name: mycontainer
+    image: myimage
+    volumeMounts:
+    - name: secrets-store-inline
+      mountPath: "/mnt/secrets-store"
+      readOnly: true
+  volumes:
+  - name: secrets-store-inline
+    csi:
+      driver: secrets-store.csi.k8s.io
+      readOnly: true
+      volumeAttributes:
+        secretProviderClass: "db-credentials"
+```
+
+In this example, the Pod `mypod` mounts a volume at `/mnt/secrets-store`. This volume is provided by the Secrets Store CSI driver and is configured to fetch secrets as defined by the `db-credentials` SecretProviderClass. The secrets fetched by the driver will be written to files in this volume. For instance, the secret `db/username` will be written to the file `/mnt/secrets-store/db_username` in the pod's filesystem.
+
+#### Helm Charts
+
+We'll use Helm charts to manage the deployment of our Conjur provider, which will simplify the deployment process and make it more reliable.
+
+The Helm chart will encapsulate the details of the deployment configuration, including the definition of deployment resources, configuration of the Conjur provider, and the necessary Kubernetes roles and role bindings.
+
+### Development Tasks
+
+The implementation of the CyberArk Conjur provider for the Kubernetes Secrets Store CSI Driver will require several tasks to be completed. These tasks can be grouped into several categories: implementation, testing, documentation, and release.
+
+#### Implementation Tasks
+
+1. **Implement the Conjur Provider**: Develop a Go-based application that uses the CyberArk Conjur API to authenticate and fetch secrets. This application should implement the `CSIDriverProviderServer` interface, which includes the `Mount` and `Version` methods. These methods will need to communicate with the CyberArk Conjur API to fetch secrets and return them to the Secrets Store CSI Driver.
+
+2. **Create the Helm Charts**: Develop the necessary Kubernetes resource definitions (Deployment, Service, etc.) and values files to manage the deployment of the Conjur provider. This Helm chart will allow users to easily deploy and configure the Conjur provider in their Kubernetes clusters.
+
+3. **Integration with CyberArk Conjur**: The logic for the provider to authenticate with the Conjur API and fetch secrets needs to be developed. This will involve using the `authn-jwt` authenticator for authentication with the workload service account token and making API calls to fetch secrets.
```

Comment:
Given that some of what we need to do here has been done before in Secrets Provider and the AuthnK8s client, how should we plan on using/repurposing that content?

We've talked before about Secrets Provider's packages being difficult to consume out-of-context - should we have a dev task to consider/address that?